File: Q(14-05-2023).java
import java.util.Scanner;
public class Main 
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        String fruits = sc.nextLine();
        String result = findFruitsLeftOver(fruits);
        System.out.println(result);
    }
    public static String findFruitsLeftOver(String fruits) 
    {
        StringBuilder sb = new StringBuilder(fruits);
        boolean isSmashPossible = true;
        while (isSmashPossible) 
        {
            isSmashPossible = false;
            for (int i = 0; i < sb.length() - 2; i++) 
            {
                if (sb.charAt(i) != sb.charAt(i+1) && sb.charAt(i+1) != sb.charAt(i+2) && sb.charAt(i) != sb.charAt(i+2)) 
                {
                    sb.delete(i, i+3);
                    isSmashPossible = true;
                    break;
                }
            }
        }
        return sb.length() > 0 ? sb.toString() : "No fruit is left over";
    }
}

File: Q(14-05-2023).py
def smash_fruits(fruits):
    i = 0
    while i + 2 < len(fruits):
        if fruits[i] != fruits[i+1] and fruits[i] != fruits[i+2] and fruits[i+1] != fruits[i+2]:
            return True
        i += 1
    return False

def find_leftover_fruits(fruits):
    while smash_fruits(fruits):
        i = 0
        while i + 2 < len(fruits):
            if fruits[i] != fruits[i+1] and fruits[i] != fruits[i+2] and fruits[i+1] != fruits[i+2]:
                fruits = fruits[:i] + fruits[i+3:]
                break
            i += 1
    if len(fruits) == 0:
        print("No fruit is left over")
    else:
        print(''.join(fruits))
        
fruits = input().strip()
find_leftover_fruits(fruits)
